host/windows/syscall.c

```diff
@@ -324,6 +326,449 @@ __declspec(noreturn) static void _panic(
+        if ((pathlen >=3 && path[0] == '/' && isalpha(path[1]) && path[2] == '/') ||
+                // /c only
+                (pathlen == 2 && path[0] =='/' && isalpha(path[1]) && path[2] == '\0'))
+        {
+            required_size = pathlen + postlen + 1;
+            wpath = (WCHAR*)(calloc(required_size, sizeof(WCHAR)));
+            if (!wpath)
+            {
+                _set_errno(OE_ENOMEM);
+                goto done;
+            }
+
+            if(!MultiByteToWideChar(
+                CP_UTF8, 0, path, -1, wpath, (int)pathlen))
+            {
+                _set_errno(_winerr_to_errno(GetLastError()));
+                free(wpath);
+                goto done;
+            }
+            if (postlen)
+            {
+                if (!MultiByteToWideChar(
+                    CP_UTF8, 0, post, -1, wpath + pathlen - 1, (int)postlen))
+                {
+                    _set_errno(_winerr_to_errno(GetLastError()));
+                    free(wpath);
+                    goto done;
+                }
+            }
+
+            wpath[0] = wpath[1];
+            wpath[1] = ':';
+        }
+        else
+        {
+            // Absolute path needs drive letter
+            required_size = pathlen + postlen + 3;
+            wpath = (WCHAR*)(calloc(required_size, sizeof(WCHAR)));
+            if (!wpath)
+            {
+                _set_errno(OE_ENOMEM);
+                goto done;
+            }
+
+            if(!MultiByteToWideChar(
+                CP_UTF8, 0, path, -1, wpath + 2, (int)pathlen))
+            {
+                _set_errno(_winerr_to_errno(GetLastError()));
+                free(wpath);
+                goto done;
+            }
+            if (postlen)
+            {
+                if(!MultiByteToWideChar(
+                    CP_UTF8, 0, post, -1, wpath + pathlen - 1, (int)postlen))
+                {
+                    _set_errno(_winerr_to_errno(GetLastError()));
+                    free(wpath);
+                    goto done;
+                }
+            }
+
+            // getdrive returns 1 for A:
+            int drive = _getdrive();
+            if (drive <= 0)
+            {
+                _set_errno(_winerr_to_errno(GetLastError()));
+                goto done;
+            }
+
+            wpath[0] = drive + 'a' - 1;
+            wpath[1] = ':';
+        }
+    }
+    else
+    {
+        // Relative path
+        WCHAR* current_dir = _wgetcwd(NULL, 0);
+        if (!current_dir)
+        {
+            _set_errno(OE_ENOMEM);
+            goto done;
+        }
+        size_t current_dir_len = wcslen(current_dir);
+
+        required_size = pathlen + current_dir_len + postlen + 1;
+        wpath = (WCHAR*)(calloc(required_size, sizeof(WCHAR)));
+        if (!wpath)
+        {
+            _set_errno(OE_ENOMEM);
+            goto done;
+        }
+
+        oe_memcpy_s(wpath, required_size * sizeof(WCHAR), current_dir, current_dir_len * sizeof(WCHAR));
```

Comment:
oe_memcpy_s has a return value. Recommend using OE_CHECK macro to check the return value for each call.